libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -262,10 +262,13 @@ bool SyntaxChecker::visit(InlineAssembly const& _inlineAssembly)
 	if (!m_useYulOptimizer)
 		return false;
 
-	if (yul::SideEffectsCollector(
-		_inlineAssembly.dialect(),
-		_inlineAssembly.operations()
-	).containsMSize())
+	if (
+		// TODO: This should be pure syntatic instead of ignoring code inside function definitions...
```

Comment:
It's probably better to move `containsMSize` to a different class.

Reply:
Probably use `ReferencesCounter` to see if `msize` is referenced is enough?

Reply:
Oh yeah, that might work! I'm still a little puzzled why the code still contains FunctionalInstructions.